woocommerce/woocommerce-gutenberg-products-block
# Fix Single Product Classic Template block not showing on the front-end

## What

In WordPress 6.4, it appears that the global `have_posts` is `false` in the context of the full site editing single product template. This breaks the Classic Template block.

In this PR, we are creating a custom query using the available id instead of relying on the global query.

This might be a temporary workaround as we are waiting to see if that's an issue that core is willing to fix, as it might affect backwards-compatibility for other vendors.

Fixes #11454 

## Testing Instructions

1. Enable WordPress 6.4.
2. Enable a block theme.
3. Go to Site Editor → Single Product template.
4. Remove the blockified template and add the “Classic Template” block.
5. Save and check the front-end.
6. Check that the template is correctly rendered on the front-end.

(Note: styling might not be the best in TT4, but that'd be a separate issue).


## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> WordPress 6.4: fixed a bug which would break sites using the Classic Template block for the Single Product template.

## Notes

This PR is a WIP as we wait the result of this conversation: p1698246828170079-slack-C029LRAU2
